### ISSUE TITLE
Pass more array values for searching

### ISSUE THREAD

**Issue by EmdevX** [jshjohnson/Choices#164]
Is there a way to pass more values to the choices for the searching purpose?
I'm asking about this because the selectize.js take all the values and there is a way to determine in options which values should work as title and data-value. It will be awesome when Choices.js will work in the similar way.

Ex.
I would like to have possibility to search through the additional values.
```
var arrayForChoices = [{
  value: 'Example Value 1',
  label: 'Example Value 1',
  additionalValues: [
    {
      someKey: 'value 1'
      someKey: 'value 2'
    }
    {
      someOtherKey: 'value 1'
      someOtherKey: 'value 2'
    }
  ]
}, {
  value: 'Example Value 2',
  label: 'Example Value 2',
  additionalValues: [
    {
      someKey: 'value 1'
      someKey: 'value 2'
    }
    {
      someOtherKey: 'value 1'
      someOtherKey: 'value 2'
    }
  ]
}]
```

**Comment by jshjohnson** [comment]
This isn't currently possible but it would be a nice feature I think. Willing to review any pull requests

**Comment by jshjohnson** [comment]
This has now been implemented into version [2.8.6](https://github.com/jshjohnson/Choices/releases/tag/v2.8.6) 🎉. See [here](https://github.com/jshjohnson/Choices#items) for adding custom props to items and [here](https://github.com/jshjohnson/Choices#choices) for choices (note: this also works with `setChoices()` etc).
